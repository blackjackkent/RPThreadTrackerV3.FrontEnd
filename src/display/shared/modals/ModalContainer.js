import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import * as selectors from '../../../infrastructure/selectors';
import * as actions from '../../../infrastructure/actions';
import columns from '../../../infrastructure/constants/columns';
import UpsertCharacterModal from './UpsertCharacterModal';
import UpsertThreadModal from './UpsertThreadModal';
import GenericConfirmationModal from './GenericConfirmationModal';
import UpsertPublicViewModal from './UpsertPublicViewModal';

const propTypes = {
	bulkThreadsToEdit: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
	bulkUntrackThreads: PropTypes.func.isRequired,
	bulkUpdateTag: PropTypes.func.isRequired,
	bulkDeleteTag: PropTypes.func.isRequired,
	characterToEdit: PropTypes.shape({
		characterName: PropTypes.string,
		urlIdentifier: PropTypes.string
	}).isRequired,
	closeBulkUntrackThreadsModal: PropTypes.func.isRequired,
	closeBulkUpdateTagModal: PropTypes.func.isRequired,
	closeBulkDeleteTagModal: PropTypes.func.isRequired,
	closeDeletePublicViewModal: PropTypes.func.isRequired,
	closeUntrackCharacterModal: PropTypes.func.isRequired,
	closeUntrackThreadModal: PropTypes.func.isRequired,
	closeUpsertCharacterModal: PropTypes.func.isRequired,
	closeUpsertPublicViewModal: PropTypes.func.isRequired,
	closeUpsertThreadModal: PropTypes.func.isRequired,
	closeDeleteAccountConfirmationModal: PropTypes.func.isRequired,
	deletePublicView: PropTypes.func.isRequired,
	isBulkUntrackThreadsModalOpen: PropTypes.bool.isRequired,
	isBulkUpdateTagModalOpen: PropTypes.bool.isRequired,
	isBulkDeleteTagModalOpen: PropTypes.bool.isRequired,
	isDeletePublicViewModalOpen: PropTypes.bool.isRequired,
	isUntrackCharacterModalOpen: PropTypes.bool.isRequired,
	isUntrackThreadModalOpen: PropTypes.bool.isRequired,
	isUpsertCharacterModalOpen: PropTypes.bool.isRequired,
	isUpsertPublicViewModalOpen: PropTypes.bool.isRequired,
	isUpsertThreadModalOpen: PropTypes.bool.isRequired,
	isDeleteAccountConfirmationModalOpen: PropTypes.bool.isRequired,
	sortedCharacters: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
	sortedTags: PropTypes.arrayOf(PropTypes.string).isRequired,
	threadToEdit: PropTypes.shape({
		userTitle: PropTypes.string
	}).isRequired,
	untrackCharacter: PropTypes.func.isRequired,
	untrackThread: PropTypes.func.isRequired,
	upsertCharacter: PropTypes.func.isRequired,
	upsertPublicView: PropTypes.func.isRequired,
	upsertThread: PropTypes.func.isRequired,
	submitUserAccountDeletion: PropTypes.func.isRequired,
	tagToEdit: PropTypes.shape({
		selectedTag: PropTypes.string,
		updatedValue: PropTypes.string
	}).isRequired,
	viewToEdit: PropTypes.shape({
		name: PropTypes.string
	}).isRequired
};

function mapStateToProps(state) {
	const { ui, characterToEdit, threadToEdit, bulkThreadsToEdit, viewToEdit, tagToEdit } = state;
	const sortedCharacters = selectors.getCharactersSortedByIdentifier(state);
	const sortedTags = selectors.getTagsSortedByTagText(state);
	return {
		isUpsertCharacterModalOpen: ui.isUpsertCharacterModalOpen,
		isUntrackThreadModalOpen: ui.isUntrackThreadModalOpen,
		isBulkUntrackThreadsModalOpen: ui.isBulkUntrackThreadsModalOpen,
		isUpsertThreadModalOpen: ui.isUpsertThreadModalOpen,
		isUntrackCharacterModalOpen: ui.isUntrackCharacterModalOpen,
		isUpsertPublicViewModalOpen: ui.isUpsertPublicViewModalOpen,
		isDeletePublicViewModalOpen: ui.isDeletePublicViewModalOpen,
		isBulkUpdateTagModalOpen: ui.isBulkUpdateTagModalOpen,
		isBulkDeleteTagModalOpen: ui.isBulkDeleteTagModalOpen,
		isDeleteAccountConfirmationModalOpen: ui.isDeleteAccountConfirmationModalOpen,
		characterToEdit,
		threadToEdit,
		bulkThreadsToEdit,
		viewToEdit,
		tagToEdit,
		sortedCharacters,
		sortedTags
	};
}

const ModalContainer = (props) => {
	const {
		bulkThreadsToEdit,
		bulkUntrackThreads,
		bulkUpdateTag,
		bulkDeleteTag,
		characterToEdit,
		closeBulkUntrackThreadsModal,
		closeBulkUpdateTagModal,
		closeBulkDeleteTagModal,
		closeDeletePublicViewModal,
		closeUntrackCharacterModal,
		closeUntrackThreadModal,
		closeUpsertCharacterModal,
		closeUpsertPublicViewModal,
		closeUpsertThreadModal,
		closeDeleteAccountConfirmationModal,
		deletePublicView,
		isBulkUntrackThreadsModalOpen,
		isBulkUpdateTagModalOpen,
		isBulkDeleteTagModalOpen,
		isDeletePublicViewModalOpen,
		isUntrackCharacterModalOpen,
		isUntrackThreadModalOpen,
		isUpsertCharacterModalOpen,
		isUpsertPublicViewModalOpen,
		isUpsertThreadModalOpen,
		isDeleteAccountConfirmationModalOpen,
		sortedCharacters,
		sortedTags,
		threadToEdit,
		untrackCharacter,
		untrackThread,
		upsertCharacter,
		upsertPublicView,
		upsertThread,
		viewToEdit,
		tagToEdit,
		submitUserAccountDeletion
	} = props;
	return (
		<div>
			<UpsertThreadModal
				isModalOpen={isUpsertThreadModalOpen}
				onModalClose={closeUpsertThreadModal}
				thread={threadToEdit}
				onInputChange={null}
				onFormSubmit={upsertThread}
				characters={sortedCharacters}
			/>
			<UpsertCharacterModal
				isUpsertCharacterModalOpen={isUpsertCharacterModalOpen}
				closeUpsertCharacterModal={closeUpsertCharacterModal}
				submitUpsertCharacter={upsertCharacter}
				characterToEdit={characterToEdit}
			/>
			<UpsertPublicViewModal
				isUpsertPublicViewModalOpen={isUpsertPublicViewModalOpen}
				submitUpsertPublicView={upsertPublicView}
				closeUpsertPublicViewModal={closeUpsertPublicViewModal}
				viewToEdit={viewToEdit}
				characters={sortedCharacters}
				tags={sortedTags}
				columns={columns}
			/>
			<GenericConfirmationModal
				isModalOpen={isUntrackThreadModalOpen}
				submitCallback={untrackThread}
				submitButtonText="Untrack"
				closeCallback={closeUntrackThreadModal}
				closeButtonText="Cancel"
				data={threadToEdit}
				headerText="Confirm Thread Untracking"
				bodyText={
					<span>
						Are you sure you want to untrack <strong>{threadToEdit.userTitle}</strong>?
					</span>
				}
			/>
			<GenericConfirmationModal
				isModalOpen={isBulkUntrackThreadsModalOpen}
				submitCallback={bulkUntrackThreads}
				submitButtonText="Untrack"
				closeCallback={closeBulkUntrackThreadsModal}
				closeButtonText="Cancel"
				data={bulkThreadsToEdit}
				headerText="Confirm Thread Untracking"
				bodyText={`Are you sure you want to untrack ${bulkThreadsToEdit.length} threads?`}
			/>
			<GenericConfirmationModal
				isModalOpen={isUntrackCharacterModalOpen}
				submitCallback={untrackCharacter}
				submitButtonText="Untrack"
				closeCallback={closeUntrackCharacterModal}
				closeButtonText="Cancel"
				data={characterToEdit}
				headerText="Confirm Character Untracking"
				bodyText={
					<span>
						Are you sure you want to untrack{' '}
						<strong>
							{characterToEdit.characterName
								? characterToEdit.characterName
								: characterToEdit.urlIdentifier}
						</strong>
						? This will also untrack all threads associated with this character.
					</span>
				}
			/>
			<GenericConfirmationModal
				isModalOpen={isDeletePublicViewModalOpen}
				submitCallback={deletePublicView}
				submitButtonText="Delete"
				closeCallback={closeDeletePublicViewModal}
				closeButtonText="Cancel"
				data={viewToEdit}
				headerText="Confirm Public View Deletion"
				bodyText={
					<span>
						Are you sure you want to delete <strong>{viewToEdit.name}</strong>?
					</span>
				}
			/>
			<GenericConfirmationModal
				isModalOpen={isBulkUpdateTagModalOpen}
				submitCallback={bulkUpdateTag}
				submitButtonText="Update Tag"
				closeCallback={closeBulkUpdateTagModal}
				closeButtonText="Cancel"
				data={tagToEdit}
				headerText="Confirm Updated Tag Value"
				bodyText={
					<span>
						Are you sure you want to change the tag{' '}
						<strong>{tagToEdit.selectedTag}</strong> to{' '}
						<strong>{tagToEdit.updatedValue}</strong> on all your threads?
					</span>
				}
			/>
			<GenericConfirmationModal
				isModalOpen={isBulkDeleteTagModalOpen}
				submitCallback={bulkDeleteTag}
				submitButtonText="Delete Tag"
				closeCallback={closeBulkDeleteTagModal}
				closeButtonText="Cancel"
				data={tagToEdit}
				headerText="Confirm Deleted Tag Value"
				bodyText={
					<span>
						Are you sure you want to remove the tag{' '}
						<strong>{tagToEdit.selectedTag}</strong> from all your threads?
					</span>
				}
			/>
			<GenericConfirmationModal
				isModalOpen={isDeleteAccountConfirmationModalOpen}
				submitCallback={submitUserAccountDeletion}
				submitButtonText="Confirm Deletion"
				closeCallback={closeDeleteAccountConfirmationModal}
				closeButtonText="Cancel"
				data={null}
				headerText="Really Delete Account?"
				bodyText={
					<span>
						Are you sure you want to delete your account?{' '}
						<strong>This action cannot be undone.</strong> You will be unable to access
						any tracked characters or threads in the future, unless you create a new
						account and re-add them.
					</span>
				}
			/>
		</div>
	);
};

ModalContainer.propTypes = propTypes;
export default connect(mapStateToProps, {
	bulkUntrackThreads: actions.bulkUntrackThreads,
	bulkUpdateTag: actions.bulkUpdateTag,
	bulkDeleteTag: actions.bulkDeleteTag,
	closeBulkUntrackThreadsModal: actions.closeBulkUntrackThreadsModal,
	closeBulkUpdateTagModal: actions.closeBulkUpdateTagModal,
	closeBulkDeleteTagModal: actions.closeBulkDeleteTagModal,
	closeDeletePublicViewModal: actions.closeDeletePublicViewModal,
	closeUntrackCharacterModal: actions.closeUntrackCharacterModal,
	closeUntrackThreadModal: actions.closeUntrackThreadModal,
	closeUpsertCharacterModal: actions.closeUpsertCharacterModal,
	closeUpsertPublicViewModal: actions.closeUpsertPublicViewModal,
	closeUpsertThreadModal: actions.closeUpsertThreadModal,
	closeDeleteAccountConfirmationModal: actions.closeDeleteAccountConfirmationModal,
	deletePublicView: actions.deletePublicView,
	submitUserAccountDeletion: actions.submitUserAccountDeletion,
	untrackCharacter: actions.untrackCharacter,
	untrackThread: actions.untrackThread,
	upsertCharacter: actions.upsertCharacter,
	upsertPublicView: actions.upsertPublicView,
	upsertThread: actions.upsertThread
})(ModalContainer);
