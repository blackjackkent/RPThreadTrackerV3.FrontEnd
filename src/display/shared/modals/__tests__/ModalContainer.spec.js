// #region imports
import React from 'react';
import { shallowWithState } from '~/testhelpers/helpers.unit';
import ModalContainer from '../ModalContainer';
// #endregion imports

// #region mocks
jest.mock('../../../../infrastructure/actions', () => ({}));
jest.mock('../../../../infrastructure/selectors', () => ({
	getCharactersSortedByIdentifier: () => [
		{
			characterId: 1
		},
		{
			characterId: 2
		}
	],
	getTagsSortedByTagText: () => ['tag1', 'tag2', 'tag3']
}));
jest.mock('../../../../infrastructure/constants/columns', () => ({
	TEST_COLUMN: {
		key: 'testKey',
		name: 'Test Value'
	}
}));
jest.mock('../UpsertCharacterModal', () => 'UpsertCharacterModal');
jest.mock('../UpsertThreadModal', () => 'UpsertThreadModal');
jest.mock('../GenericConfirmationModal', () => 'GenericConfirmationModal');
jest.mock('../UpsertPublicViewModal', () => 'UpsertPublicViewModal');
// #endregion mocks

const createTestProps = (propOverrides) => ({
	bulkUntrackThreads: jest.fn(),
	bulkUpdateTag: jest.fn(),
	bulkDeleteTag: jest.fn(),
	closeBulkUntrackThreadsModal: jest.fn(),
	closeBulkUpdateTagModal: jest.fn(),
	closeBulkDeleteTagModal: jest.fn(),
	closeDeleteAccountConfirmationModal: jest.fn(),
	closeDeletePublicViewModal: jest.fn(),
	closeUntrackCharacterModal: jest.fn(),
	closeUntrackThreadModal: jest.fn(),
	closeUpsertCharacterModal: jest.fn(),
	closeUpsertPublicViewModal: jest.fn(),
	closeUpsertThreadModal: jest.fn(),
	deletePublicView: jest.fn(),
	isBulkUntrackThreadsModalOpen: true,
	isDeleteAccountConfirmationModalOpen: true,
	isDeletePublicViewModalOpen: true,
	isUntrackCharacterModalOpen: true,
	isUntrackThreadModalOpen: true,
	isUpsertCharacterModalOpen: true,
	isUpsertPublicViewModalOpen: true,
	isUpsertThreadModalOpen: true,
	untrackCharacter: jest.fn(),
	untrackThread: jest.fn(),
	upsertCharacter: jest.fn(),
	upsertPublicView: jest.fn(),
	upsertThread: jest.fn(),
	submitUserAccountDeletion: jest.fn(),
	...propOverrides
});

const createTestState = (stateOverrides) => ({
	ui: {
		isUpsertCharacterModalOpen: true,
		isUntrackThreadModalOpen: true,
		isBulkUntrackThreadsModalOpen: true,
		isUpsertThreadModalOpen: true,
		isUntrackCharacterModalOpen: true,
		isUpsertPublicViewModalOpen: true,
		isDeleteAccountConfirmationModalOpen: true,
		isDeletePublicViewModalOpen: true,
		isBulkUpdateTagModalOpen: true,
		isBulkDeleteTagModalOpen: true
	},
	characterToEdit: {
		characterId: 3,
		characterName: 'My Test Character'
	},
	threadToEdit: {
		threadId: 5,
		userTitle: 'My Test Thread'
	},
	bulkThreadsToEdit: [
		{
			threadId: 1
		},
		{
			threadId: 2
		}
	],
	tagToEdit: {
		selectedTag: 'Test Tag',
		updatedValue: 'Test Tag Edited'
	},
	viewToEdit: {
		id: 10,
		name: 'My Test View'
	},
	...stateOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const state = createTestState();
			const jsx = <ModalContainer {...props} />;
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot with missing character name', () => {
			const props = createTestProps();
			const state = createTestState({
				characterToEdit: {
					id: 12,
					urlIdentifier: 'my-test-character'
				}
			});
			const jsx = <ModalContainer {...props} />;
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
	});
});
