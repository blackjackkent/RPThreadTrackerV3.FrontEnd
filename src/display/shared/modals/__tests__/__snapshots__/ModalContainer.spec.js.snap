// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<div>
  <UpsertThreadModal
    characters={
      Array [
        Object {
          "characterId": 1,
        },
        Object {
          "characterId": 2,
        },
      ]
    }
    closeUpsertThreadModal={[MockFunction]}
    isUpsertThreadModalOpen={true}
    submitUpsertThread={[MockFunction]}
    threadToEdit={
      Object {
        "threadId": 5,
        "userTitle": "My Test Thread",
      }
    }
  />
  <UpsertCharacterModal
    characterToEdit={
      Object {
        "characterId": 3,
        "characterName": "My Test Character",
      }
    }
    closeUpsertCharacterModal={[MockFunction]}
    isUpsertCharacterModalOpen={true}
    submitUpsertCharacter={[MockFunction]}
  />
  <UpsertPublicViewModal
    characters={
      Array [
        Object {
          "characterId": 1,
        },
        Object {
          "characterId": 2,
        },
      ]
    }
    closeUpsertPublicViewModal={[MockFunction]}
    columns={
      Object {
        "TEST_COLUMN": Object {
          "key": "testKey",
          "name": "Test Value",
        },
      }
    }
    isUpsertPublicViewModalOpen={true}
    submitUpsertPublicView={[MockFunction]}
    tags={
      Array [
        "tag1",
        "tag2",
        "tag3",
      ]
    }
    viewToEdit={
      Object {
        "id": 10,
        "name": "My Test View",
      }
    }
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to untrack 
        <strong>
          My Test Thread
        </strong>
        ?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "threadId": 5,
        "userTitle": "My Test Thread",
      }
    }
    headerText="Confirm Thread Untracking"
    isModalOpen={true}
    submitButtonText="Untrack"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText="Are you sure you want to untrack 2 threads?"
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Array [
        Object {
          "threadId": 1,
        },
        Object {
          "threadId": 2,
        },
      ]
    }
    headerText="Confirm Thread Untracking"
    isModalOpen={true}
    submitButtonText="Untrack"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to untrack
         
        <strong>
          My Test Character
        </strong>
        ? This will also untrack all threads associated with this character.
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "characterId": 3,
        "characterName": "My Test Character",
      }
    }
    headerText="Confirm Character Untracking"
    isModalOpen={true}
    submitButtonText="Untrack"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to delete 
        <strong>
          My Test View
        </strong>
        ?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "id": 10,
        "name": "My Test View",
      }
    }
    headerText="Confirm Public View Deletion"
    isModalOpen={true}
    submitButtonText="Delete"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to change the tag
         
        <strong>
          Test Tag
        </strong>
         to
         
        <strong>
          Test Tag Edited
        </strong>
         on all your threads?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "selectedTag": "Test Tag",
        "updatedValue": "Test Tag Edited",
      }
    }
    headerText="Confirm Updated Tag Value"
    isModalOpen={true}
    submitButtonText="Update Tag"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to remove the tag
         
        <strong>
          Test Tag
        </strong>
         from all your threads?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "selectedTag": "Test Tag",
        "updatedValue": "Test Tag Edited",
      }
    }
    headerText="Confirm Deleted Tag Value"
    isModalOpen={true}
    submitButtonText="Delete Tag"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to delete your account?
         
        <strong>
          This action cannot be undone.
        </strong>
         You will be unable to access any tracked characters or threads in the future, unless you create a new account and re-add them.
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={null}
    headerText="Really Delete Account?"
    isModalOpen={true}
    submitButtonText="Confirm Deletion"
    submitCallback={[MockFunction]}
  />
</div>
`;

exports[`rendering snapshots should render valid snapshot with missing character name 1`] = `
<div>
  <UpsertThreadModal
    characters={
      Array [
        Object {
          "characterId": 1,
        },
        Object {
          "characterId": 2,
        },
      ]
    }
    closeUpsertThreadModal={[MockFunction]}
    isUpsertThreadModalOpen={true}
    submitUpsertThread={[MockFunction]}
    threadToEdit={
      Object {
        "threadId": 5,
        "userTitle": "My Test Thread",
      }
    }
  />
  <UpsertCharacterModal
    characterToEdit={
      Object {
        "id": 12,
        "urlIdentifier": "my-test-character",
      }
    }
    closeUpsertCharacterModal={[MockFunction]}
    isUpsertCharacterModalOpen={true}
    submitUpsertCharacter={[MockFunction]}
  />
  <UpsertPublicViewModal
    characters={
      Array [
        Object {
          "characterId": 1,
        },
        Object {
          "characterId": 2,
        },
      ]
    }
    closeUpsertPublicViewModal={[MockFunction]}
    columns={
      Object {
        "TEST_COLUMN": Object {
          "key": "testKey",
          "name": "Test Value",
        },
      }
    }
    isUpsertPublicViewModalOpen={true}
    submitUpsertPublicView={[MockFunction]}
    tags={
      Array [
        "tag1",
        "tag2",
        "tag3",
      ]
    }
    viewToEdit={
      Object {
        "id": 10,
        "name": "My Test View",
      }
    }
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to untrack 
        <strong>
          My Test Thread
        </strong>
        ?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "threadId": 5,
        "userTitle": "My Test Thread",
      }
    }
    headerText="Confirm Thread Untracking"
    isModalOpen={true}
    submitButtonText="Untrack"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText="Are you sure you want to untrack 2 threads?"
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Array [
        Object {
          "threadId": 1,
        },
        Object {
          "threadId": 2,
        },
      ]
    }
    headerText="Confirm Thread Untracking"
    isModalOpen={true}
    submitButtonText="Untrack"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to untrack
         
        <strong>
          my-test-character
        </strong>
        ? This will also untrack all threads associated with this character.
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "id": 12,
        "urlIdentifier": "my-test-character",
      }
    }
    headerText="Confirm Character Untracking"
    isModalOpen={true}
    submitButtonText="Untrack"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to delete 
        <strong>
          My Test View
        </strong>
        ?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "id": 10,
        "name": "My Test View",
      }
    }
    headerText="Confirm Public View Deletion"
    isModalOpen={true}
    submitButtonText="Delete"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to change the tag
         
        <strong>
          Test Tag
        </strong>
         to
         
        <strong>
          Test Tag Edited
        </strong>
         on all your threads?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "selectedTag": "Test Tag",
        "updatedValue": "Test Tag Edited",
      }
    }
    headerText="Confirm Updated Tag Value"
    isModalOpen={true}
    submitButtonText="Update Tag"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to remove the tag
         
        <strong>
          Test Tag
        </strong>
         from all your threads?
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={
      Object {
        "selectedTag": "Test Tag",
        "updatedValue": "Test Tag Edited",
      }
    }
    headerText="Confirm Deleted Tag Value"
    isModalOpen={true}
    submitButtonText="Delete Tag"
    submitCallback={[MockFunction]}
  />
  <GenericConfirmationModal
    bodyText={
      <span>
        Are you sure you want to delete your account?
         
        <strong>
          This action cannot be undone.
        </strong>
         You will be unable to access any tracked characters or threads in the future, unless you create a new account and re-add them.
      </span>
    }
    closeButtonText="Cancel"
    closeCallback={[MockFunction]}
    data={null}
    headerText="Really Delete Account?"
    isModalOpen={true}
    submitButtonText="Confirm Deletion"
    submitCallback={[MockFunction]}
  />
</div>
`;
